R/sim.R

```diff
@@ -148,6 +148,14 @@ sim <- function (ode = NULL,
   if(!is.null(lagtime)) {
     regimen <- apply_lagtime(regimen, lagtime, parameters, attr(ode, "cmt_mapping"))
   }
+  if(!is.null(attr(ode, "dose")$duration_scale)) {
+    regimen <- apply_duration_scale(
+      regimen,
+      attr(ode, "dose")$duration_scale,
+      parameters,
+      attr(ode, "cmt_mapping")
```

Comment:
```suggestion
      parameters
```
this argument is no longer necessary